src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1669,12 +1672,17 @@
                         (filter expired?)
                         sort)
         drop-one (fn [table]
-                   (update-lock-status status-key inc)
-                   (try!
-                    (jdbc/do-commands
-                     (format "drop table if exists %s cascade" table))
-                    (finally
-                      (update-lock-status status-key dec))))
+                   (let [{current :version} (sutils/db-metadata)
+                          reduced-locking pg14-db]
+                     (update-lock-status status-key inc)
+                     (try!
+                       (jdbc/do-commands
+                         (if (neg? (compare current reduced-locking))
+                           (format "drop table if exists %s cascade" table)
+                           (let [base-table-name (str/replace table #"_[^_]+$" "")]
```

Comment:
There might be a better way to get this? I didn't see a utility method in partitioning.clj.

Reply:
nm; I hadn't noticed that table-prefix was available.